sycl/test/graph/graph-memory.cpp

```diff
@@ -0,0 +1,44 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+#include <CL/sycl.hpp>
+#include <iostream>
+
+#include <sycl/ext/oneapi/experimental/graph.hpp>
+
+int main() {
+
+  sycl::property_list properties{
+      sycl::property::queue::in_order{},
+      sycl::ext::oneapi::property::queue::lazy_execution{}};
+
+  sycl::queue q{sycl::gpu_selector_v, properties};
+
+  sycl::ext::oneapi::experimental::command_graph g;
+
+  const size_t n = 1000;
+  float *x;
+  //void*& x = vec;
+  
+  auto a = g.add_malloc((void*&)x,n,sycl::usm::alloc::shared);
```

Comment:
I think we should try and avoid forcing the user to do this cast if possible. Though I think the only way to let the user avoid the cast would be to template `add_malloc`, otherwise they either need to pass in a `void*` themselves or do the cast shown here.

Reply:
Yes agreed. The initial draft was not really user friendly in this scenario. Changing it to a templated version for now. We can always add other versions for the sake of completeness.